src/main/java/com/icthh/xm/ms/entity/util/DatabaseUtil.java

```diff
@@ -32,4 +39,20 @@ public static void createSchema(DataSource dataSource, String name) {
             throw new RuntimeException("Can not connect to database", e);
         }
     }
+
+    public static void runAfterTransaction(XmEntity xmEntity, Consumer<XmEntity> task) {
+        TransactionSynchronizationManager.registerSynchronization(new TransactionSynchronizationAdapter() {
+            @Override
+            public void afterCompletion(int status) {
+                super.afterCompletion(status);
+                if (status == STATUS_COMMITTED) {
+                    try {
+                        task.accept(xmEntity);
+                    } catch (Exception e) {
+                        log.error("Error save entity to elastic {}", xmEntity, e);
```

Comment:
i thinks there could be not only elastic related Consumer. 
propose to change log message to more generic: 
`Error in afterCompletion stage during transaction synchronization for entity: {}`